src/Commands/GenerateSitemap.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+namespace Portable\FilaCms\Commands;
+
+use Illuminate\Console\Command;
+use Illuminate\Support\Facades\Route;
+use Portable\FilaCms\Models\Form;
+use Portable\FilaCms\Facades\FilaCms;
+use Str;
+
+class GenerateSitemap extends Command
+{
+    protected $signature = 'fila-cms:generate-sitemap';
+
+    protected $description = 'Generate sitemap';
+
+    protected $exemptedWords = [
+        '_',
+        'storybook',
+        'horizon',
+        'livewire',
+        'two-factor-challenge',
+        'sanctum',
+        'filament',
+        'reset-password',
+        'media/{media}',
+        's/{slug}',
+        'link/{type}/{slug}',
+    ];
+
+    protected $exemptedMiddleware = [
+        'filament-actions',
+        'Illuminate\Session\Middleware\AuthenticateSession',
+        'Laravel\Pulse\Http\Middleware\Authorize',
+        'auth',
+        'auth:sanctum',
+        'auth:web'
+    ];
+
+    public function handle()
+    {
+        $routes = Route::getRoutes();
+
+        $filteredRoutes = [];
+        $this->generateSlugs('');
+
+        foreach ($routes as $key => $route) {
+            $shouldInclude = true;
+
+            // only GET routes
+            if (collect($route->methods())->search('GET') !== false) {
+                foreach ($this->exemptedWords as $word) {
+                    if (Str::of($route->uri)->startsWith($word)) {
+                        $shouldInclude = false;
+                    }
+                }
+
+                foreach ($this->exemptedMiddleware as $middleware) {
+                    if (collect($route->gatherMiddleware())->search($middleware) !== false) {
+                        $shouldInclude = false;
+                    }
+                }
+
+                if ($shouldInclude) {
+                    if (Str::of($route->uri)->contains('{slug}') == false) {
+                        $filteredRoutes[] = [url($route->uri), now()->subDays(7)];
+                    }
+                }
+            }
+        }
+
+        // add other slugs
+        $slugs = $this->generateSlugs();
+        $filteredRoutes = array_merge($filteredRoutes, $slugs);
+
+        $sitemap = $this->generateSitemap($filteredRoutes);
+
+        file_put_contents(public_path('/sitemap.xml'), $sitemap);
+    }
+
+    protected function generateSlugs()
+    {
+        $models = FilaCms::getRawContentModels();
+        unset($models["Portable\FilaCms\Models\Page"]);
+
+        $records = [];
+
+        foreach ($models as $model => $resource) {
```

Comment:
The first two "where's" shouldn't be necessary?  The default query scope should take care of giving you only published things

Reply:
Oooohhhh yeah totally forgot about our default scope LMAO